synapse/event_auth.py

```diff
@@ -181,16 +201,46 @@ def check_auth_rules_for_event(
                 "room appears to have unsupported version %s" % (room_version_prop,),
             )
 
-        logger.debug("Allowing! %s", event)
         return
 
-    auth_dict = {(e.type, e.state_key): e for e in auth_events}
-
     # 3. If event does not have a m.room.create in its auth_events, reject.
     creation_event = auth_dict.get((EventTypes.Create, ""), None)
     if not creation_event:
         raise AuthError(403, "No create event in auth events")
 
+
+def check_auth_rules_for_event(
```

Comment:
Perhaps `check_state_dependent_auth_rules_for_event`?

Reply:
I think this is [d731b22](https://github.com/matrix-org/synapse/pull/13065/commits/d731b2220581c9bbd0cbca47802d189a547506cc) ?